testing/docker.go

```diff
@@ -23,10 +23,11 @@ import (
 )
 
 func NewDockerContainer(t testing.TB, image string, env []string, cmd []string) (*DockerContainer, error) {
-	c, err := dockerclient.NewClientWithOpts()
+	c, err := dockerclient.NewClientWithOpts(dockerclient.FromEnv)
 	if err != nil {
 		return nil, err
 	}
+	c.NegotiateAPIVersion(context.TODO())
```

Comment:
This could be passed as an option to `NewClientWithOpts` after updating to a more recent client that has the  `WithAPIVersionNegotiation` option (added in https://github.com/moby/moby/pull/39032), but that's only a small improvement, and I wanted to keep the diff small

Reply:
Let's use the `dockerclient.WithAPIVersionNegotiation` option instead if it's not too much work to update the docker client version we're currently pinned to.